plugins/src/webrtcsink/imp.rs

```diff
@@ -1716,7 +1719,7 @@ impl WebRTCSink {
             }
         })?;
 
-        element.emit_by_name::<()>("new-webrtcbin", &[&peer_id, &webrtcbin]);
+        element.emit_by_name::<()>("consumer-added", &[&peer_id, &webrtcbin]);
```

Comment:
so, I see you refactored the consumer removal code path to emit the signal without the lock, should we move that code to the end of the function and add a `drop (state)` before?

Reply:
Good catch! Fixed.